### ISSUE TITLE
mailmail raises an exception instead of giving an error message

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3339]
|<img alt="doko's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| doko reported|
|-|-|
|Trac ID|trac#3339|
|Type|defect|
|Created|2008-07-08 13:42:50Z|
|Branch|https://github.com/twisted/twisted/tree/mailmail-usage-3339|

Forwarded from https://bugs.launchpad.net/bugs/148203

The man page says the addresses are optional.

```
Traceback (most recent call last):
  File "/usr/bin/mailmail", line 23, in <module>
    mailmail.run()
  File "/usr/lib/python2.5/site-packages/twisted/mail/scripts/mailmail.py", line 338, in run
    o = parseOptions(sys.argv[1:])
  File "/usr/lib/python2.5/site-packages/twisted/mail/scripts/mailmail.py", line 164, in parseOptions
    raise ValueError, "No recipients specified"
ValueError: No recipients specified
```

<details><summary>Searchable metadata</summary>

```
trac-id__3339 3339
type__defect defect
reporter__doko doko
priority__normal normal
milestone__ 
branch__branches_mailmail_usage_3339 branches/mailmail-usage-3339
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1215524570000000 1215524570000000
changetime__1215702455000000 1215702455000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="doko's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|doko commented|
|-|-|

wrong ubuntu bug number, forwarded from
https://bugs.launchpad.net/bugs/148203

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [24247]) Branching to 'mailmail-usage-3339'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set owner to @exarkun|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [24258]) Merge mailmail-usage-3339

Author: exarkun
Reviewer: therve
Fixes: [#3339](https://github.com/twisted/twisted/issues/3339)

Change `mailmail` so that it reports usage errors without a traceback; also
add some unit tests for its option parser.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
